src/pymmcore_widgets/mda/_core_mda.py

```diff
@@ -88,11 +87,24 @@ def _on_sys_config_loaded(self) -> None:
     def value(self) -> MDASequence:
         """Set the current state of the widget."""
         val = super().value()
+
+        # if the z plan is relative, and there are no stage positions, add the current
+        # stage position as the relative starting one
+        if val.z_plan and val.z_plan.is_relative and not val.stage_positions:
+            val = val.replace(stage_positions=[self._get_current_stage_position()])
```

Comment:
just a thought... might this not be better dealt with on the consumer side?  For example, if the mda engine receives an MDAsequence that wants a relative Z plan but provides no stage positions, then *it* would figure out what the current position is.